key/src/ed25519/private.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::Public;
+use primitives::H512;
+use sodiumoxide::crypto::sign::{gen_keypair, SecretKey, SECRETKEYBYTES};
+use std::ops::Deref;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Private(SecretKey);
```

Comment:
I think it's better to add a comment that `SecretKey` clears memory when it is dropped to inform others who want to replace the underlying type.

Reply:
Ok, I added a comment like "The inner type SecretKey clears its memory when it is dropped"